### PR TITLE
Update rendering.md: make documentation about managing state more discoverable.

### PR DESCRIPTION
**Pull request by konrad-jamrozik**
Added links to various parts of documentation explaining the different options of managing and propagating state updates across multiple components. I found this documentation to be not very discoverable, and only after several more hours of reading I encountered the relevant docs. Hopefully with this change the next person will save some time :)

### CODE REVIEW

File: aspnetcore/blazor/components/rendering.md [hunk 1]
```diff
@@ -420,7 +420,11 @@ The UI might involve:
...
+One way to deal with this scenario is to provide a *state management* class, often as a dependency injection (DI) service, injected into multiple components. When one component calls a method on the state manager, the state manager raises a C# event that's then received by an independent component. For an example of this, see [In-memory state container service](https://learn.microsoft.com/en-us/aspnet/core/blazor/state-management?view=aspnetcore-7.0&pivots=server#in-memory-state-container-service-server) ([WASM equivalent](https://learn.microsoft.com/en-us/aspnet/core/blazor/state-management?view=aspnetcore-7.0&pivots=webassembly#in-memory-state-container-service-server)) section of State Management article.
+
+Another option you have is to [pass data across a component hierarchy](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/cascading-values-and-parameters?view=aspnetcore-7.0#pass-data-across-a-component-hierarchy) using cascading values and paramterers.
```
Comment (konrad-jamrozik) [inline aspnetcore/blazor/components/rendering.md:425]:
```suggestion
Another option you have is to [pass data across a component hierarchy](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/cascading-values-and-parameters?view=aspnetcore-7.0#pass-data-across-a-component-hierarchy) using cascading values and parameters.
```

File: aspnetcore/blazor/components/rendering.md [hunk 2]
```diff
@@ -420,7 +420,11 @@ The UI might involve:
 1. Changing some state.
 1. Rerendering a completely different component that isn't a descendant of the component receiving the event.
 
-One way to deal with this scenario is to provide a *state management* class, often as a dependency injection (DI) service, injected into multiple components. When one component calls a method on the state manager, the state manager raises a C# event that's then received by an independent component.
+One way to deal with this scenario is to provide a *state management* class, often as a dependency injection (DI) service, injected into multiple components. When one component calls a method on the state manager, the state manager raises a C# event that's then received by an independent component. For an example of this, see [In-memory state container service](https://learn.microsoft.com/en-us/aspnet/core/blazor/state-management?view=aspnetcore-7.0&pivots=server#in-memory-state-container-service-server) ([WASM equivalent](https://learn.microsoft.com/en-us/aspnet/core/blazor/state-management?view=aspnetcore-7.0&pivots=webassembly#in-memory-state-container-service-server)) section of State Management article.
```
Comment (konrad-jamrozik) [inline aspnetcore/blazor/components/rendering.md:423]:
```suggestion
One way to deal with this scenario is to provide a *state management* class, often as a dependency injection (DI) service, injected into multiple components. When one component calls a method on the state manager, the state manager raises a C# event that's then received by an independent component. For an example of this, see [In-memory state container service](https://learn.microsoft.com/en-us/aspnet/core/blazor/state-management?pivots=server#in-memory-state-container-service-server) ([WASM equivalent](https://learn.microsoft.com/en-us/aspnet/core/blazor/state-management?pivots=webassembly#in-memory-state-container-service-server)) section of State Management article.
```

File: aspnetcore/blazor/components/rendering.md [hunk 3]
```diff
@@ -420,7 +420,11 @@ The UI might involve:
...
+One way to deal with this scenario is to provide a *state management* class, often as a dependency injection (DI) service, injected into multiple components. When one component calls a method on the state manager, the state manager raises a C# event that's then received by an independent component. For an example of this, see [In-memory state container service](https://learn.microsoft.com/en-us/aspnet/core/blazor/state-management?pivots=server#in-memory-state-container-service-server) ([WASM equivalent](https://learn.microsoft.com/en-us/aspnet/core/blazor/state-management?pivots=webassembly#in-memory-state-container-service-server)) section of State Management article.
+
+Another option you have is to [pass data across a component hierarchy](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/cascading-values-and-parameters?view=aspnetcore-7.0#pass-data-across-a-component-hierarchy) using cascading values and parameters.
```
Comment (konrad-jamrozik) [inline aspnetcore/blazor/components/rendering.md:425]:
```suggestion
Another option you have is to [pass data across a component hierarchy](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/cascading-values-and-parameters#pass-data-across-a-component-hierarchy) using cascading values and parameters.
```

File: aspnetcore/blazor/components/rendering.md [hunk 4]
```diff
@@ -420,7 +420,11 @@ The UI might involve:
...
+One way to deal with this scenario is to provide a *state management* class, often as a dependency injection (DI) service, injected into multiple components. When one component calls a method on the state manager, the state manager raises a C# event that's then received by an independent component. For an example of this, see [In-memory state container service](https://learn.microsoft.com/en-us/aspnet/core/blazor/state-management?pivots=server#in-memory-state-container-service-server) ([WASM equivalent](https://learn.microsoft.com/en-us/aspnet/core/blazor/state-management?pivots=webassembly#in-memory-state-container-service-server)) section of State Management article.
+
+Another option you have is to [pass data across a component hierarchy](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/cascading-values-and-parameters#pass-data-across-a-component-hierarchy) using cascading values and parameters.
+
+You can also pass state updates by [binding across more than two components](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/data-binding?view=aspnetcore-7.0#bind-across-more-than-two-components) using data bindings.
```
Comment (konrad-jamrozik) [inline aspnetcore/blazor/components/rendering.md:427]:
```suggestion
You can also pass state updates by [binding across more than two components](https://learn.microsoft.com/en-us/aspnet/core/blazor/components/data-binding#bind-across-more-than-two-components) using data bindings.
```

**Comment by guardrex** [comment]
Hello @konrad-jamrozik ... I'll take a look at this as soon as I can. Don't sweat the broken links and build failure. We have special linking rules that I'll take care of if we take this PR forward after review.

**Comment by dnfadmin** [comment]
[CLA assistant check](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=27983) <br/>Thank you for your submission, we really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=27983) before we can accept your contribution.<br/><br/>:x: konrad-jamrozik [sign now](https://cla.dotnetfoundation.org/dotnet/AspNetCore.Docs?pullRequest=27983)<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla.dotnetfoundation.org/check/dotnet/AspNetCore.Docs?pullRequest=27983) it.</sub>

**Comment by guardrex** [comment]
@konrad-jamrozik ... You'll need to sign the CLA if you agree with its terms in order for us to accept the PR.

Thanks for the recommended changes. I agree that the cross-links are a great idea here. I made a list out of them and used our XREF linking to solve the build warnings.

**Comment by konrad-jamrozik** [comment]
@guardrex thank you, much appreciated! :)

**Review by guardrex — APPROVED** [review]

**Pull request merged**